dsgrid/dataset/dataset_schema_handler_standard.py

```diff
@@ -40,10 +51,123 @@ def get_unique_dimension_rows(self):
+        lk_df = lookup_df or self._load_data_lookup
+
+        if query_context is not None:
+            lk_df, ld_df = self._prefilter_dataset(query_context, lk_df, ld_df)
+
+        lk_df = self._remap_dimension_columns(lk_df)
+        ld_df = self._remap_dimension_columns(
+            ld_df,
+            # Some pivot columns may have been removed.
+            pivot_columns=set(ld_df.columns).intersection(self.get_pivot_dimension_columns()),
+        )
+
+        if query_context is not None:
+            ld_df = self._process_metric_aggregations(query_context, ld_df)
+
+        ld_df = ld_df.join(lk_df.drop("fraction"), on="id").drop("id")
+
+        # Map the time here because some columns may have been collapsed above.
+        ld_df = self._convert_time_dimension(ld_df)
+
+        # All dataset columns need to be in the same order.
+        pivot_columns = sorted(
+            set(ld_df.columns).intersection(self.get_pivot_dimension_columns_mapped_to_project())
+        )
+        dim_columns = sorted(set(ld_df.columns).intersection([x.value for x in DimensionType]))
+        covered = set(pivot_columns + dim_columns)
+        remaining = sorted((x for x in ld_df.columns if x not in covered))
+        final_columns = remaining + dim_columns + pivot_columns
+        return ld_df.select(*final_columns)
+
+    def _prefilter_dataset(self, context: QueryContext, lk_df, ld_df):
+        # TODO: prefilter time
+        if context.model.project.drop_dimensions:
+            columns = [x.value for x in context.model.project.drop_dimensions]
+            lk_df = lk_df.drop(*columns)
+
+        for dim_type, project_record_ids in context.iter_record_ids_by_dimension_type():
+            dataset_mapping = self._get_dataset_to_project_mapping_records(dim_type)
+            if dataset_mapping is None:
+                dataset_record_ids = project_record_ids
+            else:
+                dataset_record_ids = (
+                    dataset_mapping.withColumnRenamed("from_id", "dataset_record_id")
+                    .join(
+                        project_record_ids,
+                        on=dataset_mapping.to_id == project_record_ids.id,
+                    )
+                    .select("dataset_record_id")
+                    .distinct()
+                    .withColumnRenamed("dataset_record_id", "id")
+                )
+            if dim_type == self.get_pivot_dimension_type():
+                # Drop columns that don't match requested project record IDs.
+                cols_to_keep = {x.id for x in dataset_record_ids.collect()}
+                cols_to_drop = set(self.get_pivot_dimension_columns()).difference(cols_to_keep)
+                if cols_to_drop:
+                    ld_df = ld_df.drop(*cols_to_drop)
+            else:
+                # Drop rows that don't match requested project record IDs.
+                tmp = dataset_record_ids.withColumnRenamed("id", "dataset_record_id")
+                lk_df = lk_df.join(
```

Comment:
Is it possible for ld_df to contain more than just time and pivoted dimension?

Reply:
Not to my knowledge. Is that a bad assumption?

Reply:
yeah, looks like standard schema handler is hard-coded to include only time, pivoted dim, and id.